mm2src/coins/utxo/bch.rs

```diff
@@ -309,6 +352,255 @@ impl BchCoin {
         )?;
         Ok(slp_address)
     }
+
+    async fn tx_from_storage_or_rpc<T: TxHistoryStorage>(
+        &self,
+        tx_hash: &H256Json,
+        storage: &T,
+    ) -> Result<UtxoTx, MmError<GetTxDetailsError<T::Error>>> {
+        let tx_hash_as_bytes = BytesJson::new(tx_hash.0.to_vec());
+        let tx_bytes = match storage.tx_bytes_from_cache(self.ticker(), &tx_hash_as_bytes).await? {
+            Some(tx_bytes) => tx_bytes,
+            None => {
+                let tx_bytes = self.as_ref().rpc_client.get_transaction_bytes(tx_hash).compat().await?;
+                storage
+                    .add_tx_to_cache(self.ticker(), &tx_hash_as_bytes, &tx_bytes)
+                    .await?;
+                tx_bytes
+            },
+        };
+        let tx = deserialize(tx_bytes.0.as_slice())?;
+        Ok(tx)
+    }
+
+    /// Returns multiple details by tx hash if token transfers also occurred in the transaction
+    pub async fn transaction_details_with_token_transfers<T: TxHistoryStorage>(
+        &self,
+        tx_hash: &H256Json,
+        block_height_and_time: Option<BlockHeightAndTime>,
+        storage: &T,
+    ) -> Result<Vec<TransactionDetails>, MmError<GetTxDetailsError<T::Error>>> {
```

Comment:
The idea is that we have even more than two transfers in one transaction. That's why I preferred to return `Vec` and implement storage in a way that it either inserts all of them or none (in a single transaction in SQLite for example). 
Also, this method doesn't guarantee that it will always return `slp_tx`, so this will have to be `Option<TransactionDetails>` making it a bit more complicated to handle.